src/errors/http-error-handler.ts

```diff
@@ -6,14 +6,19 @@ async function httpResponseErrorHandler(response: Response): Promise<Response> {
   if (!response.ok) {
     let err
     try {
+      // If it is not possible to parse the return body it means there is none
+      // In which case it is a communication error with the Meilisearch instance
       err = await response.json()
     } catch (e: any) {
+      // Not sure on how to test this part of the code.
```

Comment:
There are two handlers, one directly after the fetch command to fetch any http errors, and one other after all the handling the response as the response needs to be parsed in JSON (it is a blob) and it can go wrong.


```js
function httpErrorHandler(
  response: FetchError,
  stack?: string,
  url?: string
): Promise<void> {
  if (response.type !== 'MeiliSearchApiError') {
    throw new MeiliSearchCommunicationError(
      response.message,
      response,
      url,
      stack
    )
  }
  throw response
}
```

Its a bit weird. I tried to find a better way to handle the errors but I couldn't find one that works as intended

Reply:
Yep I got that point, but I mean, since you raised this error `new Error('fake error message')` in your test I assume that `fake error message` will be treated as a text response, not a json response, in other words, will fall into the non-json handler, correct?